src/Git.hs

```diff
@@ -146,6 +147,23 @@ data TagResult
 data FetchWithTags = WithTags | NoTags
   deriving stock (Eq, Show)
 
+data GitIntegrationFailure 
+  = MergeFailed
+  | RebaseFailed
+  | WrongFixups 
+  deriving (Show, Eq)
+
+instance FromJSON GitIntegrationFailure where
```

Comment:
Do we need these instances because it's part of hoff's persistent state? In that case, we might need some extra care with deploying. @joris-burgers I think you had a change in the persistence format as well when adding order, right? Do you remember the details?

Reply:
If you need to change the persistent state, there are 2 options:
- Make the parser backwards compatible with the previous state
- Just delete the state and redownload it.

The first one is harder to do, but in all other applications, we usually prefer this. In this case, we can probably delete the state once it is quiet and let it reinstate it. We will lose some information that way, but luckily for Hoff, we can get most of the state back automatically. Things we will lose are the merge queue, so we need to reapprove all PRs that were commanded to be merged, but were not yet merged. If we do this at a time where nobody merged, it should not be a problem.

Reply:
Also a drive-by in this case, can we not derive the `Aeson` instances here? That would be much less error-prone.

Reply:
We do need these instances as they are required for the IntegrationStatus instances in Project.hs.

Reply:
>  Also a drive-by in this case, can we not derive the Aeson instances here? That would be much less error-prone.

Well, or more error-prone, as we've seen with the failed hoff deploy just now. Deriving instances for serialization is a double-edged sword.

Reply:
> > Also a drive-by in this case, can we not derive the Aeson instances here? That would be much less error-prone.
> 
> Well, or more error-prone, as we've seen with the failed hoff deploy just now. Deriving instances for serialization is a double-edged sword.

True, but in this case we are implementing both the `FromJSON` and `ToJSON` so a failed deploy will happen either way, deriving vs manual writing doesn't make a difference I think, so under the assumption that we need the instance, my preference is to derive them, to make sure that they stay in sync and no typos are introduced or cases forgotten (either now or in the future).